queue/priority_queue.go

```diff
@@ -94,6 +107,7 @@ func (items *priorityItems) insert(item Item) {
 type PriorityQueue struct {
 	waiters     waiters
 	items       priorityItems
+	itemMap     map[Item]bool
```

Comment:
If I'm reading this correctly, `itemMap` is used to check presence only. If that's the case, you can make it

```go
itemMap map[Item]struct{}
```

and then use this pattern:

```go
// add item to map
itemMap[i] = struct{}{}

// remove item
delete(itemMap, i)

// check presence
_, ok := itemMap[i]
```

This pattern reduces the memory consumption slightly, but more important, I think it clarifies the purpose of `itemMap`. Thoughts?